R/dbplyr.R

```diff
@@ -0,0 +1,282 @@
+#' @include DBI.R
+NULL
+
+temporaryErrorMessage <- paste0("You cannot use `temporary = TRUE`",
+  "when using a Pool object, since temporary tables are local to a ",
+  "connection, and there's no guarantee you'll get the same ",
+  "connection back next time. You must either create a permanent ",
+  "table, or checkout a connection from `pool` directly with ",
+  "`con <- poolCheckout(pool)`, and then release the connection ",
+  "back to the pool when you're finished (`poolReturn(con)`).")
+
+# --- These generics are set in dplyr (not database-specific)
+#' @export
+copy_to.Pool <- function(dest, df, name = deparse(substitute(df)),
+  overwrite = FALSE, ...) {
+    db_con <- poolCheckout(dest)
```

Comment:
I now know how to convert 

```R
db_con <- poolCheckout(dest)
on.exit(poolReturn(db_con))
```

to 

```R
db_con <- localPoolCheckout(dest)
```

with the `on.exit()` being automatically added. 

It's up to you if you'd prefer that API.